leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/LeshanClientDemo.java

```diff
@@ -312,20 +321,100 @@ public static void main(final String[] args) {
+            String mastersaltStr = cl.getOptionValue("mastersalt");
+            String idcontextStr = cl.getOptionValue("idcontext");
+            String senderidStr = cl.getOptionValue("senderid");
+            String recipientidStr = cl.getOptionValue("recipientid");
+            String aeadStr = cl.getOptionValue("aead");
+            String hkdfStr = cl.getOptionValue("hkdf");
+
+            if (mastersecretStr == null) {
+                System.err.println("The OSCORE master secret must be indicated");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            // Set salt to empty string if not indicated
+            if (mastersaltStr == null) {
+                mastersaltStr = "";
+            }
+
+            if (idcontextStr != null) {
+                System.err.println("The OSCORE ID Context parameter is not yet supported");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            if (senderidStr == null) {
+                System.err.println("The OSCORE Sender ID must be indicated");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            // Check that the Sender ID matches the client name
+            // http://www.openmobilealliance.org/release/LightweightM2M/V1_1-20180710-A/HTML-Version/OMA-TS-LightweightM2M_Transport-V1_1-20180710-A.html#5-5-5-0-555-Endpoint-Client-Name
+            if (!Arrays.equals(endpoint.getBytes(), Hex.decodeHex(senderidStr.toCharArray()))) {
+                System.err.println(
+                        "The endpoint client name (" + endpoint + ") must be equal to the OSCORE Sender ID (use '"
+                                + Hex.encodeHexString(endpoint.getBytes()).toUpperCase() + "')");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            if (recipientidStr == null) {
+                System.err.println("The OSCORE Recipient ID must be indicated");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            // Parse AEAD Algorithm (set default if not indicated)
+            if (aeadStr == null) {
+                aeadStr = "AES_CCM_16_64_128";
+            }
+
+            int aeadInt;
+            if (aeadStr.equals("AES_CCM_16_64_128") || aeadStr.equals("10")) {
+                aeadInt = 10;
+            } else {
+                System.err.println("The AEAD algorithm set is not supported (AES_CCM_16_64_128 recommended)");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            // Parse HKDF Algorithm (set default if not indicated)
+            if (hkdfStr == null) {
+                hkdfStr = "HKDF_HMAC_SHA_256";
+            }
+
+            int hkdfInt;
+            if (hkdfStr.equals("HKDF_HMAC_SHA_256") || hkdfStr.equals("-10")) {
+                hkdfInt = -10;
```

Comment:
same here ?

Reply:
Did the same thing as for the AEAD algorithm, the HKDF algorithm is now parsed into a COSE AlgorithmID and not compared it to fixed values. Done in commit 76fa1268a8f8262eb39008693f37a75069173ba2